common/scala/src/main/scala/whisk/core/connector/Message.scala

```diff
@@ -53,7 +54,8 @@ case class ActivationMessage(
     activationId: ActivationId,
     activationNamespace: EntityPath,
     content: Option[JsObject],
-    cause: Option[ActivationId] = None)
+    cause: Option[ActivationId] = None,
+    traceMetadata: Option[SpanMetadata] = None)
```

Comment:
> I think this feature can continue to be developed but doesn't that's to impact the core.

Hopefully not in a separate branch, but in the master branch; any performance related work should be derived from the insight uncovered by tracing. My hope is that we can automatically generate reports from the work by @markusthoemmes and @tysonnorris on performance testing, by exporting a report every time we run these performance tests in Jenkins. 

I think the easiest solution right now would be to use a _Feature Flag_ , in a similar way we did with ContainerPool. 

Otherwise, if we want to dynamically load some libraries, I think we can do this with `ZipkinLogging` but b/c the Kafka message needs to be decorated with Tracing Information, and `Message.scala` is part of the `core`, we would have to compile all Zipkin libs with the core anyway. Unless we build a way to decorate the `Message`.

Any suggestions here ?